src/main/java/com/csfrequency/firebase/devicestore/FirebaseDeviceStore.java

```diff
@@ -142,70 +154,45 @@ public void unsubscribe() {
         subscribed = false;
     }
 
-    private void deleteDevice(final String userId) {
+    private Task<Void> deleteDevice(final String userId) {
         final DocumentReference docRef = userRef(userId);
 
-        firestore.runTransaction(new Transaction.Function<Void>() {
+        return docRef.get().continueWithTask(new Continuation<DocumentSnapshot, Task<Void>>() {
```

Comment:
This can be further simplified if you are okay with creating a new document if the document does not yet exists (which contradicts my last feedback, but you had extra code to address this case at the time). 

I believe you could simply do:

```
docRef.set(map(DEVICES_FIELD, map(getDeviceId(), FieldValue.detele()), SetOptions.merge())
```